unlock-app/src/hooks/useProvider.ts

```diff
@@ -35,18 +35,17 @@ export const useProvider = (config: any) => {
   const [encryptedPrivateKey, setEncryptedPrivateKey] = useState<
     any | undefined
   >(undefined)
-  const { getStorage, setStorage, clearStorage } = useAppStorage()
+  const { setStorage, clearStorage } = useAppStorage()
   const { addNetworkToWallet } = useAddToNetwork(account)
 
   useEffect(() => {
-    if (!getStorage('account') && account) {
+    if (account) {
       setStorage('account', account)
     }
-
-    if (!getStorage('network') && network) {
+    if (network) {
       setStorage('network', network)
     }
-  }, [account, network])
+  }, [account, network, setStorage])
```

Comment:
This was a bug where local storage account won't be updated when you switch directly using your wallet.